### PR TITLE
[BPK-2460] Add other variants of the component

### PR DESCRIPTION
**Pull request by georgegillams**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [ ] `README.md` (another PR)
+ [ ] Tests (another PR)
+ [x] Docs here https://github.com/Skyscanner/backpack-docs/pull/73

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

Default
Small
smallWithIcons
withIcons
withoutUnderline

### CODE REVIEW

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m [hunk 1]
```diff
@@ -89,16 +92,29 @@ - (void)setSelectedColor:(UIColor *_Nullable)selectedColor {
...
+        [self updateStyle];
+        [self updateIconStyle];
+    }
+}
+
 #pragma mark - Private
 
 - (void)updateStyle {
     BPKFontStyle fontStyle = BPKFontStyleTextBase;
+    if(self.size == BPKHorizontalNavigationSizeSmall) {
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m]: Spacing

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m [hunk 2]
```diff
@@ -111,8 +127,55 @@ - (UIColor *)contentColor {
     return BPKColor.gray300;
 }
 
+- (void)setIconColor {
+    if (self.iconView != nil) {
+        self.iconView.tintColor = [self contentColor];
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m]: Can we use property syntax for properties i.e. `self.contentColor` not `[self contentColor]`?

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m [hunk 3]
```diff
@@ -111,8 +127,55 @@ - (UIColor *)contentColor {
...
+    }
+}
+
+- (void)updateIconStyle {
+    if (self.iconView != nil) {
+        [self.iconView removeFromSuperview];
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m]: This seems redundant if we still have an icon

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m [hunk 4]
```diff
@@ -111,8 +127,55 @@ - (UIColor *)contentColor {
...
+
+    BPKIconSize size = self.size == BPKHorizontalNavigationSizeDefault ? BPKIconSizeLarge : BPKIconSizeSmall;
+    self.iconView = [[BPKIconView alloc] initWithIconName:self.definition.iconName size:size];
+
+    self.iconView.translatesAutoresizingMaskIntoConstraints = NO;
+    self.titleLabel.translatesAutoresizingMaskIntoConstraints = NO;
+
+    [self addSubview:self.iconView];
+
+    [NSLayoutConstraint activateConstraints:@[
```
Comment (k0nserv) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigationItem.m]: `UIButton` natively supports showing images I think we should prefer this to modifying its view hierarchy

File: Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m [hunk 5]
```diff
@@ -121,8 +122,8 @@ - (void)adjustBarConstraintsForView:(UIView *)view {
         [NSLayoutConstraint deactivateConstraints:self.barConstraints];
     }
     self.barConstraints = @[
-        [self.barView.leadingAnchor constraintEqualToAnchor:view.leadingAnchor constant:-BPKSpacingBase],
```
Comment (georgegillams) [inline Backpack/HorizontalNavigation/Classes/BPKHorizontalNavigation.m]: Gonna handle this in the item component itself as that'll give a better tap area

**Pull request merged**